woocommerce/woocommerce
# Fix broken db update button on homescreen

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This fixes a bug in the inbox note to update database. The note previously relied on a mechanism to update the db from the current URL the notice is on. However this is not compatible with the home screen, as WooCommerce Admin's will route the URL with JS instead of performing a server load. To get around this, we use the WP dashboard URL (which WCA won't try to route, and implement a redirect back to the original page.


https://github.com/woocommerce/woocommerce/assets/9312929/b452df0e-709a-48ac-a9d3-62ea065a7676



Closes https://github.com/woocommerce/woocommerce/issues/47172 .




### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Go to WooCommerce > Home in admin
2. See the notice for the update (if it doesn't appear on dev sites, it may require rewinding the db version, e.g `wp option set woocommerce_db_version 8.9.0`
3. Perform the update
5. Confirm the homescreen reloads and shows an updating or thanks message.

To further test around this, reset the database version again and try from other screens.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
